systems/plants/RigidBody.m

```diff
@@ -316,7 +316,9 @@ function dofnum(obj)
       % class. (maybe I should move it up to there?)
       fn = fieldnames(body.param_bindings);
       for i=1:length(fn)
-        body.(fn{i}) = double(subs(body.param_bindings.(fn{i}),poly,pval));
+%         body.(fn{i}) = double(subs(body.param_bindings.(fn{i}),poly,pval));
+        body.(fn{i}) = subs(body.param_bindings.(fn{i}),poly,pval);
+        if (deg(body.(fn{i}))==0) body.(fn{i}) = double(body.(fn{i}));end
```

Comment:
Not in my experience.  This line was a recommendation from Russ, fwiw.

-Andy S.

On Thu, Jul 30, 2015 at 11:15 PM, Twan Koolen <notifications@github.com>
wrote:

> In systems/plants/RigidBody.m
> <https://github.com/RobotLocomotion/drake/pull/1212#discussion_r35943878>:
>
> > @@ -316,7 +316,9 @@ function dofnum(obj)
> >        % class. (maybe I should move it up to there?)
> >        fn = fieldnames(body.param_bindings);
> >        for i=1:length(fn)
> > -        body.(fn{i}) = double(subs(body.param_bindings.(fn{i}),poly,pval));
> > +%         body.(fn{i}) = double(subs(body.param_bindings.(fn{i}),poly,pval));
> > +        body.(fn{i}) = subs(body.param_bindings.(fn{i}),poly,pval);
> > +        if (deg(body.(fn{i}))==0) body.(fn{i}) = double(body.(fn{i}));end
>
> does this break non-msspoly inputs?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RobotLocomotion/drake/pull/1212/files#r35943878>.
>